### PR TITLE
fix: ImmutableSet converted to List for Impersonated Credentials

### LINKED ISSUE #731: BigQuery/Storage - ImpersonatedCredentials fails when trying to get Service
#### Environment details

1. Specify the API at the beginning of the title. For example, "BigQuery: ...").
   General, Core, and Other are also allowed as types
2. OS type and version: MAC OS Big Sur 11.5.2
3. Java version: 8
4. google-auth-library version(s): 1.1.0

#### Steps to reproduce

  1. Impersonate the account : gcloud config set auth/impersonate_service_account <serviceaccount>
  2. Set application default credentials : gcloud auth application-default login

#### Code example

```java
import com.google.auth.oauth2.GoogleCredentials;
import com.google.cloud.bigquery.BigQuery;

GoogleCredentials credentials = ComputeEngineCredentials.getApplicationDefault();
BigQuery bigquery = BigQueryOptions.newBuilder().setProjectId(projectId).setCredentials(credentials).build()
				.getService();
// example
```

#### Stack trace
```
Caused by: java.lang.ClassCastException: class com.google.common.collect.SingletonImmutableSet cannot be cast to class java.util.List (com.google.common.collect.SingletonImmutableSet is in unnamed module of loader 'app'; java.util.List is in module java.base of loader 'bootstrap')
	at com.google.auth.oauth2.ImpersonatedCredentials.createScoped(ImpersonatedCredentials.java:367)
	at com.google.cloud.ServiceOptions.getScopedCredentials(ServiceOptions.java:596)
	at com.google.cloud.http.HttpTransportOptions.getHttpRequestInitializer(HttpTransportOptions.java:145)
	at com.google.cloud.bigquery.spi.v2.HttpBigQueryRpc.<init>(HttpBigQueryRpc.java:105)
	at com.google.cloud.bigquery.BigQueryOptions$DefaultBigQueryRpcFactory.create(BigQueryOptions.java:56)
	at com.google.cloud.bigquery.BigQueryOptions$DefaultBigQueryRpcFactory.create(BigQueryOptions.java:50)
	at com.google.cloud.ServiceOptions.getRpc(ServiceOptions.java:560)
	at com.google.cloud.bigquery.BigQueryOptions.getBigQueryRpcV2(BigQueryOptions.java:125)
	at com.google.cloud.bigquery.BigQueryImpl.<init>(BigQueryImpl.java:249)
	at com.google.cloud.bigquery.BigQueryOptions$DefaultBigQueryFactory.create(BigQueryOptions.java:46)
	at com.google.cloud.bigquery.BigQueryOptions$DefaultBigQueryFactory.create(BigQueryOptions.java:40)
	at com.google.cloud.ServiceOptions.getService(ServiceOptions.java:540)
```

#### External references such as API reference guides

- NA

#### Any additional information below

-

### PR DESCRIPTION
**Pull request by dpanangat**
As part of BigQueryOptions or StorageOptions the SCOPES is defined as an ImmutableSet.
When using an Impersonated Account, it fails when trying to get service, as the SCOPES are immutableSet and the previous code tries to cast it to List directly and fails.
Adding a fix for the same.

Fixes #731 ☕️

### CODE REVIEW

**Comment by dpanangat** [comment]
@googlebot I signed it!

**Review by chingor13 — CHANGES_REQUESTED** [review]
Thanks for submitting this. Can you add a simple test for this failure so that we don't regress in the future?

File: oauth2_http/javatests/com/google/auth/oauth2/ImpersonatedCredentialsTest.java [hunk 1]
```diff
@@ -60,12 +60,9 @@
 import java.nio.charset.Charset;
 import java.security.PrivateKey;
 import java.text.SimpleDateFormat;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Calendar;
-import java.util.Date;
-import java.util.List;
-import java.util.Map;
+import java.util.*;
```
Comment (TimurSadykov) [inline oauth2_http/javatests/com/google/auth/oauth2/ImpersonatedCredentialsTest.java:63]: Please keep individual imports

File: oauth2_http/javatests/com/google/auth/oauth2/ImpersonatedCredentialsTest.java [hunk 2]
```diff
@@ -107,11 +104,13 @@
           + "CJzdWIiOiIxMDIxMDE1NTA4MzQyMDA3MDg1NjgifQ.redacted";
   public static final String ACCESS_TOKEN = "1/MkSJoj1xsli0AccessToken_NKPY2";
 
+  private static final Set<String> IMMUTABLE_SCOPES = ImmutableSet.of("scope1","scope2");
+
```
Comment (TimurSadykov) [inline oauth2_http/javatests/com/google/auth/oauth2/ImpersonatedCredentialsTest.java:108]: nit: please remove this empty line

File: oauth2_http/javatests/com/google/auth/oauth2/ImpersonatedCredentialsTest.java [hunk 3]
```diff
@@ -107,11 +104,13 @@
           + "CJzdWIiOiIxMDIxMDE1NTA4MzQyMDA3MDg1NjgifQ.redacted";
   public static final String ACCESS_TOKEN = "1/MkSJoj1xsli0AccessToken_NKPY2";
 
+  private static final Set<String> IMMUTABLE_SCOPES = ImmutableSet.of("scope1","scope2");
```
Comment (TimurSadykov) [inline oauth2_http/javatests/com/google/auth/oauth2/ImpersonatedCredentialsTest.java:107]: please rename to IMMUTABLE_SCOPES_SET

File: oauth2_http/javatests/com/google/auth/oauth2/ImpersonatedCredentialsTest.java [hunk 4]
```diff
@@ -107,11 +104,13 @@
...
+
   private static final String PROJECT_ID = "project-id";
   public static final String IMPERSONATED_CLIENT_EMAIL =
       "impersonated-account@iam.gserviceaccount.com";
   private static final List<String> SCOPES =
```
Comment (TimurSadykov) [inline oauth2_http/javatests/com/google/auth/oauth2/ImpersonatedCredentialsTest.java:112]: Please rename to IMMUTABLE_SCOPES_LIST

**Review by TimurSadykov — CHANGES_REQUESTED** [review]
Hi, sorry for delay, some minor comments and we can get it done. Thanks!

File: oauth2_http/javatests/com/google/auth/oauth2/ImpersonatedCredentialsTest.java (hunk 2 above)
Comment (dpanangat) [reply oauth2_http/javatests/com/google/auth/oauth2/ImpersonatedCredentialsTest.java]: removed this line

**Review by TimurSadykov — APPROVED** [review]
LGTM

**Comment by TimurSadykov** [comment]
@dpanangat hey, thanks for updates, please fix the linter failure and we should be good to go. To do that you need to run the "mvn com.coveo:fmt-maven-plugin:format" command and push resulted format changes.

**Review by chingor13 — APPROVED** [review]

**Pull request merged**